pandas-stubs/core/frame.pyi

```diff
@@ -698,16 +698,44 @@ class DataFrame(NDFrame, OpsMixin, _GetItemHack):
         self,
         expr: _str,
         *,
+        parser: Literal["pandas", "python"] = ...,
+        engine: Literal["python", "numexpr"] | None = ...,
+        local_dict: dict[_str, Any] | None = ...,
+        global_dict: dict[_str, Any] | None = ...,
+        resolvers: list[Mapping] | None = ...,
+        level: int = ...,
+        target: object | None = ...,
         inplace: Literal[True],
-        **kwargs: Any,  # TODO: make more precise https://github.com/pandas-dev/pandas-stubs/issues/1173
+    ) -> None: ...
+    @overload
+    def query(
+        self,
+        expr: _str,
+        *,
+        inplace: Literal[True],
+        **kwargs: Any,
     ) -> None: ...
```

Comment:
Not clear why you need this overload when the previous overload covers it?

Reply:
This test would raise a mypy/pyright error without the other overload.

https://github.com/pandas-dev/pandas-stubs/blob/d19ac8994dd69d4a1663310b45ce6cd288bd2cac/tests/test_frame.py#L532-L536

Reply:
My take on this is that we shouldn't support the `kwargs` being passed in a `dict`, because they can't be type-checked. The docs say what `kwargs` are acceptable, via the `eval()` link, so let's not include this test (and the overloads supporting `kwargs`